src/java.base/share/classes/java/lang/String.java

```diff
@@ -3645,8 +3645,6 @@ static String join(String prefix, String suffix, String delimiter, String[] elem
      */
     public static String join(CharSequence delimiter,
             Iterable<? extends CharSequence> elements) {
-        Objects.requireNonNull(delimiter);
-        Objects.requireNonNull(elements);
         var delim = delimiter.toString();
         var elems = new String[8];
         int size = 0;
```

Comment:
Can you add a `// implicit null check` before the loop? Otherwise, the null check may be forgotten that future code change may early return before the null check here is triggered.